File: packages/javascript/lib/word/randMyth.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Creatures and figures out of myth and folklore.
 *
 * `randWord({ theme: 'myth' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randMyth({ language: 'ko', count: 3 }); // ['용', '봉황', '구미호']
 * randMyth({ language: 'en', count: 3 }); // ['Dragon', 'Phoenix', 'Griffin']
 */
export function randMyth(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Myth words with the language and theme each one came from.
 *
 * @example
 * randMyth({ language: 'ko', output: 'detail' });
 * // [{ word: '용', language: 'ko', theme: 'myth' }]
 */
export function randMyth(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randMyth(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('myth', options);
}

File: packages/dart/lib/src/word/rand_myth.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Creatures and figures out of myth and folklore.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.myth] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randMyth(language: WordLanguage.ko, count: 3); // [용, 봉황, 구미호]
/// randMyth(language: WordLanguage.en, count: 3); // [Dragon, Phoenix, Griffin]
/// ```
List<String> randMyth({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.myth,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);
